google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/JdbcResultSetMetaData.java

```diff
@@ -0,0 +1,205 @@
+
+  private final ResultSet spannerResultSet;
+  private final Statement statement;
+
+  JdbcResultSetMetaData(JdbcResultSet jdbcResultSet, Statement statement) {
+    Preconditions.checkNotNull(jdbcResultSet);
+    this.spannerResultSet = jdbcResultSet.spanner;
+    this.statement = statement;
+  }
+
+  @Override
+  public boolean isClosed() {
+    return false;
+  }
+
+  @Override
+  public int getColumnCount() throws SQLException {
+    return spannerResultSet.getColumnCount();
+  }
+
+  @Override
+  public boolean isAutoIncrement(int column) throws SQLException {
+    return false;
+  }
+
+  @Override
+  public boolean isCaseSensitive(int column) throws SQLException {
+    int type = getColumnType(column);
+    return type == Types.NVARCHAR || type == Types.BINARY;
+  }
+
+  @Override
+  public boolean isSearchable(int column) throws SQLException {
+    return true;
+  }
+
+  @Override
+  public boolean isCurrency(int column) throws SQLException {
+    return false;
+  }
+
+  @Override
+  public int isNullable(int column) throws SQLException {
+    return columnNullableUnknown;
+  }
+
+  @Override
+  public boolean isSigned(int column) throws SQLException {
+    int type = getColumnType(column);
+    return type == Types.DOUBLE || type == Types.BIGINT;
+  }
+
+  @Override
+  public int getColumnDisplaySize(int column) throws SQLException {
+    int colType = getColumnType(column);
+    switch (colType) {
+      case Types.ARRAY:
+        return 50;
+      case Types.BOOLEAN:
+        return 5;
+      case Types.BINARY:
+        int binaryLength = getPrecision(column);
+        return binaryLength == 0 ? 50 : binaryLength;
+      case Types.DATE:
+        return 10;
+      case Types.DOUBLE:
+        return 14;
+      case Types.BIGINT:
+        return 10;
+      case Types.NVARCHAR:
+        int length = getPrecision(column);
+        return length == 0 ? 50 : length;
+      case Types.TIMESTAMP:
+        return 16;
+      default:
+        return 10;
+    }
+  }
+
+  @Override
+  public String getColumnLabel(int column) throws SQLException {
+    return spannerResultSet.getType().getStructFields().get(column - 1).getName();
+  }
+
+  @Override
+  public String getColumnName(int column) throws SQLException {
+    return spannerResultSet.getType().getStructFields().get(column - 1).getName();
+  }
+
+  @Override
+  public String getSchemaName(int column) throws SQLException {
+    return statement.getConnection().getSchema();
+  }
+
+  @Override
+  public int getPrecision(int column) throws SQLException {
+    int colType = getColumnType(column);
+    switch (colType) {
+      case Types.BOOLEAN:
+        return 1;
+      case Types.DATE:
+        return 10;
+      case Types.DOUBLE:
+        return 14;
+      case Types.BIGINT:
+        return 10;
+      case Types.TIMESTAMP:
+        return 24;
+      default:
+        // Not fixed size, we could try to get it from INFORMATION_SCHEMA
+        return 50;
```

Comment:
from here, guessing it's default precision?

Reply:
For text columns, this method should return the size of the column in number of characters. Cloud Spanner does not return that information, meaning that we could only get it by parsing the SQL statement that has been used to create the result set in order to figure out which column it is, and then fetch it from INFORMATION_SCHEMA. I've added some more documentation to make that clearer.